network/proxy.md

```diff
@@ -6,68 +6,154 @@ keywords: network, networking, proxy, client
+
+```json
+{
+ "proxies": {
+   "default": {
+     "httpProxy": "http://192.168.1.12:3128",
+     "httpsProxy": "https://192.168.1.12:3129",
+     "noProxy": "*.test.example.com,.example.org,127.0.0.0/8"
+   }
+ }
+}
+```
+
+> **Warning**
+>
+> Proxy settings may contain sensitive information. For example, some proxy servers
+> require authentication information to be included in their URL, or their
+> address may expose IP-addresses or hostnames of your company's environment.
+>
+> Environment variables are stored as plain text in the container's configuration,
+> and as such can be inspected through the remote API or committed to an image
+> when using `docker commit`.
+{: .warning }
+
+The configuration becomes active after saving the file, you don't need to
+restart Docker. However, the configuration only applies to new containers and
+builds, and doesn't affect existing containers.
+
+The following table describes the available configuration parameters.
+
+| Property     | Description                                                                         |
+| :----------- | :---------------------------------------------------------------------------------- |
+| `httpProxy`  | Sets the `HTTP_PROXY` and `http_proxy` environment variables and build arguments.   |
+| `httpsProxy` | Sets the `HTTPS_PROXY` and `https_proxy` environment variables and build arguments. |
+| `ftpProxy`   | Sets the `FTP_PROXY` and `ftp_proxy` environment variables and build arguments.     |
+| `noProxy`    | Sets the `NO_PROXY` and `no_proxy` environment variables and build arguments.       |
+| `allProxy`   | Sets the `ALL_PROXY` and `all_proxy` environment variables and build arguments.     |
+
+These settings are used to configure proxy environment variables for containers
+only, and not used as proxy settings for the Docker CLI or the Docker Engine
+itself.
+Refer to the [environment variables](/engine/reference/commandline/cli/#environment-variables)
+and [configure the Docker daemon to use a proxy server](../config/daemon/systemd.md#configure-the-docker-daemon-to-use-a-proxy-server)
+sections for configuring proxy settings for the CLI and daemon.
+
+### Run containers with a proxy configuration
+
+When you run containers, proxy-related environment variables such as
+`HTTP_PROXY` reflect the values set in the configuration file.
+
+```console
+$ docker run -it alpine
+/ # echo $HTTP_PROXY
+http://192.168.1.12:3128
+```
+
+### Build with a proxy configuration
+
+When you invoke a build, proxy-related build arguments are pre-populated
+automatically, based on the proxy settings in your Docker client configuration
+file. You can access the proxy values using `ARG` instructions in your
+Dockerfile:
+
+```dockerfile
+FROM alpine
+ARG HTTPS_PROXY
+RUN apk update    # uses httpsProxy from ~/.docker/config.json
+```
+
+### Configure proxy settings per daemon
+
+The `proxies.default` key configures the proxy settings for all daemons that
```

Comment:
Slightly wondering if `proxies.default` could lead to users using that as a literal key. Perhaps we should look at slightly different phrasing.